Application/EdFi.Ods.SandboxAdmin/Services/BackgroundJobService.cs

```diff
@@ -1,45 +1,126 @@
             {
                 // refresh existing sandboxes periodically
                 if (user.Value.Sandboxes.Any(s => s.Value.Refresh))
                 {
                     // Change the recurrence to suit your needs using Cron functions or a unix CRON expressions (i.e. "* */6 * * *" = every 6 hours)
-                    RecurringJob.AddOrUpdate("RebuildSandboxes", () => _engine.RebuildSandboxes(), Cron.Daily(), TimeZoneInfo.Local);
+                    await CreateAndScheduleNewJob<RebuildSandboxesJob>("createSandboxes", "setupAdminDatabase", scheduler, "* */24 * * *");
+                }
+            }
+            
+            async Task<IJobDetail> CreateAndScheduleNewJob<T>(string idName, string idGroup, IScheduler scheduler, string cronExpression = null)
+                where T : IJob
+            {
+                var newJob = JobBuilder.Create<T>().WithIdentity(idName, idGroup).Build();
+                newJob.JobDataMap.Put("engine", _engine);
+                newJob.JobDataMap.Put("exitAfterSandboxCreation", exitAfterSandboxCreation);
+                ITrigger newTrigger;
+            
+                // If no cron expression is provided, then use a simple schedule that runs once
+                if (string.IsNullOrEmpty(cronExpression))
+                {
+                    newTrigger = TriggerBuilder.Create()
+                        .WithIdentity(idName, idGroup)
+                        .WithSimpleSchedule()
+                        .StartNow()
+                        .Build();
+                }
+                else
+                {
+                    newTrigger = TriggerBuilder.Create()
+                        .WithIdentity(idName, idGroup)
+                        .WithCronSchedule(cronExpression)
+                        .StartNow()
+                        .Build();
                 }
+
+                await scheduler.ScheduleJob(newJob, newTrigger);
+                return newJob;
+            }
+        }
+    }
+
+    [DisallowConcurrentExecution]
+    public class CreateIdentityRolesJob : IJob
+    {
+        public async Task Execute(IJobExecutionContext context)
+        {
+            await ((IInitializationEngine)context.MergedJobDataMap["engine"]).CreateIdentityRoles();
+        }
+    }
+
+    [DisallowConcurrentExecution]
+    public class CreateIdentityUsersJob : IJob
+    {
+        public async Task Execute(IJobExecutionContext context)
+        {
+            await ((IInitializationEngine)context.MergedJobDataMap["engine"]).CreateIdentityUsers();
+        }
+    }
+
+    [DisallowConcurrentExecution]
+    public class RebuildSandboxesJob : IJob
+    {
+        public async Task Execute(IJobExecutionContext context)
+        {
+            ((IInitializationEngine)context.MergedJobDataMap["engine"]).CreateSandboxes();
+        }
+    }
+
+    [DisallowConcurrentExecution]
+    public class CreateSandboxesJob : IJob
+    {
+        public async Task Execute(IJobExecutionContext context)
```

Comment:
This warning can be fixed by removing the `async` keyword and returning `Task.CompletedTask` or by returning a Task from InitializationEngine